### PR TITLE
fix raft log gc failure

### PR DESCRIPTION
**Pull request by lidezhu**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: proxy_components/engine_tiflash/src/ps_engine/ps_log_engine.rs [hunk 1]
```diff
@@ -312,7 +312,6 @@ impl PSLogEngine {
             return Ok(0);
         }
 
-        let mut raft_wb = self.log_batch(0);
```
Comment (CalvinNeo) [inline proxy_components/engine_tiflash/src/ps_engine/ps_log_engine.rs:315]: I think we can enrich the test of `test_raft_engine_wb`. We can consume `raft_wb_clean` and check if the raft states are dropped.

Comment (lidezhu) [reply proxy_components/engine_tiflash/src/ps_engine/ps_log_engine.rs:315]: OK.

Comment (lidezhu) [reply proxy_components/engine_tiflash/src/ps_engine/ps_log_engine.rs:315]: Added.

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Pull request merged**